### ISSUE TITLE
UI freezes in BlueStacks 5 Android emulator upon start

### ISSUE THREAD

**Issue by anno73** [bk138/droidVNC-NG#111]
Hi!

When activating droidVNC-NG in BlueStacks the UI freezes and no UI update happens any more regardless of pressing the home or back button. I have the impression that the buttons still work and just UI is not updated but I cannot confirm this for sure.

ADB connection is still working. This was the only way I could uninstall droidVNC-NG as I unfortunately enabled "Start on Boot".

The only way out is to restart whole BlueStacks and not enable "Start on Boot".

BlueStacks was running Android Nougat 64Bit.

Interestingly, when enabling droidVNC-NG with Start button, the warning pop up for record/projection shows up as expected but then starts to behave weird (flashing...) upon pressing allow button.

On client side I could connect with RealVNC Viewer but I only got a black window with no content.

Not sure if this is a BlueStacks or droidVNC-NG issue. On my Huawei P30 it works nicely :-)

Br, Alex.

**Comment by bk138** [comment]
A few ideas I have you could try:
* you could try with ScreenRecorder mentioned in https://github.com/bk138/droidVNC-NG#notes and see if it casues a freeze as well
* `adb logcat` output is really really helpful!

**Comment by bradaker** [comment]
Just an FYI on the one part of your report where you say it starts flashing...  I had that same thing happen when I had forgotten that another VNC server (as I was comparing multiple) was already listening on the same tcp port that I was trying to start droidVNC-NG on. And for all I know it didn't have to be another VNC server per se (just happened to be in my case). It might have done the same if just any other daemon/service had that port in use.

**Comment by anno73** [comment]
@bk138 I attached a very lengthy adb logcat dump:

[logcat.txt](https://github.com/bk138/droidVNC-NG/files/10045576/logcat.txt)

I would expect that the interesting things start at around line 5050 at ~ 01:01:00. There are jumps in timestamp. maybe ntp?

What I found out meanwhile too is, that BlueStacks has an option to select "Graphics engine mode" - Compatibility and Performance. The issues only arises when Performance is selected. With Compatibility it works. The logcat dump is when Performance was selected.

And only graphics output is locked, because I can start and stop droidVNC-NG by clicking on the not updating Start/Stop button area. Real VNC Viewer reports the connection then as terminated.

I will try ScreenRecorder later.

@bradaker I changed the port to different numbers with no noticeable effect :-) And I can connect with VNC Viewer, so I think this is not an issue.
